Add a Sign-Up page to Accounts

MelissaAutumn opened the pull request:
Fixes #455 

I've done some large re-organization here to help improve boot time, and allow us to import keycloak components into account's vue app. 

In the future it'd be nice to have a normalized folder structure (e.g. `assets/modules/accounts` and `assets/modules/keycloak`.)  While I'm able to simply just import the keycloak assets and have vite bake them into the build, I still had to copy over the i18n manually as keycloak uses property files for their locale. 

This also includes some additional improvements like the ability to turn off the base template for accounts for a full-page view, and also I've added support for a bulk importer for the newly created AllowListEntry model. 

## Changes
* Added AllowListEntry model
* Added bulk import for AllowListEntry (in admin panel)
* Adjusted KeycloakClient.import_user to send any aia action as an email
* Adjusted KeycloakClient.import_user to send over passwords
* Added slots to Keycloak's register view
* Refactored Keycloak's frontend vite to use `@kc` for local imports
* Added SignUpView that's mostly an import of Keycloak's RegisterView
* Only slightly sorry for not naming SignUpView as RegisterView
* Endured a few migraines
* Split Keycloak's base template from App.vue into BaseTemplate
* Injected primary domain to Keycloak's template via env variables
* Added a link to account's sign-up page on Keycloak's login page via env variables (if registration is disabled on kc)
* Added wait list url setting to accounts
* Disabled Sentry for dev / tests (it was getting annoying in stack traces)
* Add ruff check for rogue print statements in our codebase (you can use `noqa: T201` to suppress them per line)
* Switched to sublime-text with a language server, so added some helpers for typing within django

## Docker changes

I've moved frontend prod build, and collectstatic to the dockerfile's build step, instead of on boot. This should speed up container boots since we don't need to rebuild the frontend everytime we boot. Technically this step isn't even needed in local dev, but it's there so we can use the same container on stage/prod. 

If you experience any issues during testing wrt to docker please document and post here as part of the review. Thanks!

MelissaAutumn commented on keycloak/themes/tbpro/assets/vue/App.vue:
```diff
@@ -1,164 +1,17 @@
-      </div>
-    </div>
-  </section>
+  <base-template>
```
Simplified so we can re-use the base template without having to refactor the App.vue logic.

MelissaAutumn commented on run-tests.sh:
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+sudo docker compose exec accounts uv run manage.py test $1
```
ex/ `./run-tests.sh src.thunderbird_accounts.authentication.tests.SignUpViewTestcase.test_not_on_allowed_list`

I've switched to sublime-text with lsp-pyright and lsp-ruff, so no more fancy pycharm test runners for me 😅

davinotdavid reviewed (commented):
Super cool! Thanks! 🎉 I will give it a spin running locally and this review comment is just code-wise and it all lgtm, just one nit for now:

MelissaAutumn commented:
Haha that's odd. It's supposed to be email verification, and will look into the password matching.

MelissaAutumn commented:
Fixed the client side password/confirm password check, and added the actual action to the update account page

For future fixes see https://github.com/thunderbird/thunderbird-accounts/issues/466

davinotdavid reviewed (commented):
Nice! I can verify that the Verify Action (heh) now shows the "Verify Email" text, however I am seeing two small things in the Sign Up form:

1. `doesnt` instead of `doesn't`

<img width="500" height="152" alt="Image" src="https://github.com/user-attachments/assets/d97c6762-7f45-45e6-90bf-127a89259d0c" />

2. No `recovery email` email type validation

Adding a `type="email"` to L114 in `keycloak/themes/tbpro/assets/vue/views/RegisterView/index.vue` seems to do it

<img width="559" height="780" alt="Image" src="https://github.com/user-attachments/assets/b598719d-5c02-4e62-aafb-23e21dd36773" />
